diff --git a/dist/klinecharts-pro.js b/dist/klinecharts-pro.js
--- a/dist/klinecharts-pro.js
+++ b/dist/klinecharts-pro.js
@@ -3390,5 +3390,6 @@
     getSymbol: () => y(),
     setPeriod: M,
-    getPeriod: () => L()
+    getPeriod: () => L(),
+    getChartApi: () => n
   });
   const c1 = () => {
@@ -3876,5 +3877,5 @@
     this._container.classList.add("klinecharts-pro"), this._container.setAttribute("data-theme", t.theme ?? "light");
     const n = this;
-    H5(() => d(Dl, {
+    this._dispose = H5(() => d(Dl, {
       ref: (r) => {
         n._chartApi = r;
@@ -3958,4 +3959,11 @@
     }), this._container);
   }
+  getChartApi() {
+    return this._chartApi.getChartApi();
+  }
+  destroy() {
+    this._dispose?.();
+    this._dispose = undefined;
+  }
   setTheme(t) {
     var n;
